File: CSharp/LeaderBoard/Driver.cs
namespace TDDMicroExercises.LeaderBoard
{
    public class Driver
    {
        public Driver(string name, string country)
        {
            Name = name;
            Country = country;
        }

        public string Name { get; }
        public string Country { get; }

        public override int GetHashCode()
        {
            return Name.GetHashCode()*31 + Country.GetHashCode();
        }

        public override bool Equals(object obj)
        {
            if (this == obj)
            {
                return true;
            }
            if (!(obj is Driver))
            {
                return false;
            }
            var other = (Driver) obj;
            return Name.Equals(other.Name) && Country.Equals(other.Country);
        }
    }
}

File: php/src/Leaderboard/Driver.php
<?php

declare(strict_types=1);

namespace RacingCar\Leaderboard;

use Stringable;

class Driver implements Stringable
{
    public function __construct(
        public ?string $name,
        public string $country
    ) {
    }

    public function __toString(): string
    {
        return (string) "{$this->name} {$this->country}";
    }
}
